product_seasonality/models/seasonal_config_line.py

```diff
@@ -0,0 +1,66 @@
+# Copyright 2021 Camptocamp SA
+# @author: Julien Coux <julien.coux@camptocamp.com>
+# @author: Simone Orsi <simone.orsi@camptocamp.com>
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
+
+from odoo import _, api, fields, models
+from odoo.exceptions import ValidationError
+
+
+class SeasonalConfigLine(models.Model):
+    _name = "seasonal.config.line"
+    _description = "Product seasonal configuration Line"
+
+    seasonal_config_id = fields.Many2one(
+        string="Product seasonal configuration",
+        comodel_name="seasonal.config",
+        required=True,
+    )
+    product_id = fields.Many2one(
+        comodel_name="product.product",
+        domain=[("sale_ok", "=", True)],
+        required=True,
+    )
+    date_start = fields.Datetime(required=True)
+    date_end = fields.Datetime()
+    monday = fields.Boolean(default=True)
+    tuesday = fields.Boolean(default=True)
+    wednesday = fields.Boolean(default=True)
+    thursday = fields.Boolean(default=True)
+    friday = fields.Boolean(default=True)
+    saturday = fields.Boolean(default=True)
+    sunday = fields.Boolean(default=True)
+
+    @api.constrains("date_start", "date_end")
+    def _check_date_end(self):
+        for line in self:
+            if line.date_end and line.date_end < line.date_start:
+                raise ValidationError(
+                    _("The end date cannot be earlier than the start date.")
+                )
+
+    def is_sale_ok(self, date):
+        self.ensure_one()
+
+        weekday = date.strftime("%A").lower()
+        # check if we are on the correct day of the week
+        if not self[weekday]:
+            return False
+
+        # check if the provided date is between date_start and date_end
+        # of the seasonal conf line
+        # TODO: include limits w/ `=`?
+        if date < self.date_start or self.date_end and date > self.date_end:
```

Comment:
Seems good for me like this: if your seasonal products can be sold from the 2021-06-01, the 2021-05-31 must return False, but 2021-06-01 yes.
```suggestion
        if date < self.date_start or self.date_end and date > self.date_end:
```